src/subscription_manager/syspurposelib.py

```diff
@@ -216,5 +237,105 @@ def read_syspurpose():
             syspurpose = json.load(open(USER_SYSPURPOSE))
         except (os.error, ValueError):
             # In the event this file could not be read treat it as empty
+            if raise_on_error:
+                raise
             syspurpose = {}
     return syspurpose
+
+
+def write_syspurpose(values):
+    """
+    Write the syspurpose to the file system.
+    :param values:
+    :return:
+    """
+    if SyspurposeStore is not None:
+        sp = SyspurposeStore(USER_SYSPURPOSE)
+        sp.contents = values
+        sp.write()
+    else:
+        # Simple backup in case the syspurpose tooling is not installed.
```

Comment:
Do we really need this backup? I thought that subscription_manager package will have dependency on syspurpose. We IMHO create unnecessary duplicity of code here.